docs/user/metal/install_ipi.md

```diff
@@ -0,0 +1,264 @@
+
+* **NIC #2 - Provisioning Network**
+  * A private, non-routed network, used for PXE based provisioning.
+  * DHCP is automated for this network.
+  * Addressing for this network is currently hard coded as `172.22.0.0/24`, but
+    will be made configurable in the future.
+
+* **Out-of-band Management Network**
+  * Servers will typically have an additional NIC used by the onboard
+    management controllers (BMCs).  These BMCs must be accessible and routed to
+    the host.
+
+### Provisioning Host
+
+The installer must be run from a host that is attached to the same networks as
+the cluster, as described in the previous section.  We refer to this host as
+the *provisioning host*.  The easiest way to provide a provisioning host is to
+use one of the hosts that is intended to later become a worker node in the same
+cluster.  That way it is already connected to the proper networks.
+
+It is recommended that the provisioning host be a bare metal host, as it must be
+able to use libvirt to launch the OpenShift bootstrap VM locally.
+
+### Supported Hardware
+
+The architecture is intended to support a wide variety of hardware.  This was
+one of the reasons Ironic is used as an underlying technology.  However, so far
+development and testing has focused on PXE based provisioning using IPMI for
+out-of-band management of hosts.  Other provisioning approaches will be added,
+tested, and documented over time.
+
+## Installation Process
+
+Once an environment has been prepared according to the documented
+pre-requisites, the install process is the same as other IPI based platforms.
+
+`openshift-install create cluster`
+
+However, it is recommended to prepare an `install-config.yaml` file in advance,
+containing all of the details of the bare metal hosts to be provisioned.
+
+### Install Config
+
+The `install-config.yaml` file requires some additional details.  Most of the
+information is teaching the installer and the resulting cluster enough about
+the available hardware so that it is able to fully manage it.
+
+Here is an example `install-config.yaml` with the required `baremetal` platform
+details.
+
+**IMPORTANT NOTE:** The current install configuration for the `baremetal`
+platform should be considered experimental and still subject to change without
+backwards compatibility.  In particular, some items likely to change soon
+include:
+
+* The `image` section will get completely removed.
+
+```yaml
+apiVersion: v1beta4
+baseDomain: test.metalkube.org
+metadata:
+  name: ostest
+compute:
+- name: worker
+  replicas: 1
+controlPlane:
+  name: master
+  replicas: 3
+  platform:
+    baremetal: {}
+platform:
+  baremetal:
+    api_vip: 192.168.111.5
+    hosts:
+      - name: openshift-master-0
+        role: master
+        bmc:
+          address: ipmi://192.168.111.1:6230
+          username: admin
+          password: password
+        bootMACAddress: 00:11:07:4e:f6:68
+        hardwareProfile: default
```

Comment:
I struggle with this one, and I think you've answered it before - what are valid values for this field? How can a user find those out?

Reply:
These are the valid values: https://github.com/metal3-io/baremetal-operator/blob/master/pkg/hardware/profile.go#L48

Not sure if there's user-facing docs for this, @dhellmann would know.  Maybe we should include it here if it's not anywhere else.

Reply:
The user shouldn't really have to set the hardware profile. Have you encountered cases where setting it is required?

Reply:
@dhellmann How do derive the hardware profile? We're reusing BMO in the installer for this code, and we use the hardware profile to get the various settings needed (root device hints, etc). At the moment, the user has to specify the hardware profile in the install-config

Reply:
The operator has some basic logic to try to derive the value. If that's not sufficient for the hardware platforms we have, we need to open some bugs describing where the guessing isn't right. 

In the mean time, it sounds like we need the user-facing field until we improve the guessing. There isn't any documentation for what the profiles are, yet, so adding some basic details here makes sense.

Reply:
Unless we know of a case where we have to set this now, and we don't think that case can be quickly fixed, I wouldn't add this field - much easier to add it back later than to remove it

Reply:
You can omit it, I suppose. However, the operator's controller code isn't being executed here, this is just being used to get the values to pass to terraform. I assume for bare metal, you need to specify one of the dell profiles so root device hints are correct, don't you? Or is everyone basically using the default?

Reply:
I think most people are getting by fine just using the default.

I suspect there may be cases where we need to provide a root device hint, and this is the only way we have to do so.  It has to be configured in install-config, as this is stuff that has to get fed into Ironic from terraform.  There's no way to dynamically determine this at install time.

I'm going to document it as something that's likely to change while this is still considered experimental, though.

Reply:
I've added a note about the `hardwareProfile` setting.  Please comment further if you feel it's not sufficient.  Thanks!